src/io.js

```diff
@@ -146,3 +146,128 @@ export default function twoline2satrec(longstr1, longstr2) {
 
   return satrec;
 }
+
+/* -----------------------------------------------------------------------------
+ *
+ *                           function json2satrec
+ *
+ *  this function converts the OMM json data to variables and initializes the sgp4 
+ *    variables. several intermediate varaibles and quantities are determined. note 
+ *    that the result is a structure so multiple satellites can be processed 
+ *    simultaneously without having to reinitialize. the verification mode is an 
+ *    important option that permits quick checks of any changes to the underlying 
+ *    technical theory. this option works using a modified tle file in which the 
+ *    start, stop, and delta time values are included at the end of the second line
+ *    of data. this only works with the verification mode. the catalog mode simply 
+ *    propagates from -1440 to 1440 min from epoch and is useful when performing 
+ *    entire catalog runs.
+ *
+ *  author        : david vallado                  719-573-2600    1 mar 2001
+ *
+ *  inputs        :
+ *    jsonobj     - OMM json data
+ *    typerun     - type of run                    verification 'v', catalog 'c',
+ *                                                 manual 'm'
+ *    typeinput   - type of manual input           mfe 'm', epoch 'e', dayofyr 'd'
+ *    opsmode     - mode of operation afspc or improved 'a', 'i'
+ *    whichconst  - which set of constants to use  72, 84
+ *
+ *  outputs       :
+ *    satrec      - structure containing all the sgp4 satellite information
+ *
+ *  coupling      :
+ *    getgravconst-
+ *    days2mdhms  - conversion of days to month, day, hour, minute, second
+ *    jday        - convert day month year hour minute second into julian date
+ *    sgp4init    - initialize the sgp4 variables
+ *
+ *  references    :
+ *    norad spacetrack report #3
+ *    vallado, crawford, hujsak, kelso  2006
+ --------------------------------------------------------------------------- */
+
+/**
+ * Return a Satellite imported from OMM JSON format data.
+ *
+ * Provide the OMM JSON format data as `jsonobj`,
+ * and select which standard set of gravitational constants you want
+ * by providing `gravity_constants`:
+ *
+ * `sgp4.propagation.wgs72` - Standard WGS 72 model
+ * `sgp4.propagation.wgs84` - More recent WGS 84 model
+ * `sgp4.propagation.wgs72old` - Legacy support for old SGP4 behavior
+ *
+ * Normally, computations are made using letious recent improvements
+ * to the algorithm.  If you want to turn some of these off and go
+ * back into "afspc" mode, then set `afspc_mode` to `True`.
```

Comment:
Improved mode is currently set no matter what. Comment should be updated or the option should be added as a parameter.